src/workloads/query/QueryStatsLimited.yml

```diff
@@ -0,0 +1,175 @@
+    Parameters:
+      MaxPhase: *MaxPhase
+      Phases: [
+        13, 23, 33, 43,
+        16, 26, 36, 46,
+      ]
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: PrepopulateActor  # Prepopulate the query stats cache until it's full. This is to simulate
+                           # the steady-state scenario, which is what we're mostly interested in.
+    Parameters:
+      CacheSizeMb: *CacheSizeMb
+      MaxPhase: *MaxPhase
+      Phases: [14, 24, 34, 44]
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: RateLimitActor  # Reset the rate limit to the desired value. This needs to be kept
+                         # up-to-date with the actual rate limit.
+    Parameters:
+      MaxPhase: *MaxPhase
+      Phases: [15, 25, 35, 45]
+      RateLimitHz: *RateLimitHz
+
+# No concurrent $queryStats query, all find queries are roughly the same size.
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: FindCommandActor
+    Parameters:
+      CacheSizeMb: *CacheSizeMb
+      MaxPhase: *MaxPhase
+      Phase: 17
+      Payload: *FixedPayload
+      RateLimitHz: *RateLimitHz
+      Suffix: "Base-Fixed-RateLimited"
+
+# Concurrent HMAC-transformed $queryStats query, all find queries are roughly the same size.
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: FindCommandActor
+    Parameters:
+      CacheSizeMb: *CacheSizeMb
+      MaxPhase: *MaxPhase
+      Phase: 27
+      Payload: *FixedPayload
+      RateLimitHz: *RateLimitHz
+      Suffix: "Hmac-Fixed-RateLimited"
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: QueryStatsActor
+    Parameters:
+      CacheSizeMb: *CacheSizeMb
+      MaxPhase: *MaxPhase
+      Phase: 27
+      QueryStatsParameters: *HmacParameters
+      RateLimitHz: *RateLimitHz
+      Suffix: "Hmac-Fixed-RateLimited"
+
+# No concurrent $queryStats query, find queries vary widely in size.
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: FindCommandActor
+    Parameters:
+      CacheSizeMb: *CacheSizeMb
+      MaxPhase: *MaxPhase
+      Phase: 37
+      Payload: *RandomPayload
+      RateLimitHz: *RateLimitHz
+      Suffix: "Base-Random-RateLimited"
+
+# Concurrent HMAC-transformed $queryStats query, find queries vary widely in size.
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: FindCommandActor
+    Parameters:
+      CacheSizeMb: *CacheSizeMb
+      MaxPhase: *MaxPhase
+      Phase: 47
+      Payload: *RandomPayload
+      RateLimitHz: *RateLimitHz
+      Suffix: "Hmac-Random-RateLimited"
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: QueryStatsActor
+    Parameters:
+      CacheSizeMb: *CacheSizeMb
+      MaxPhase: *MaxPhase
+      Phase: 47
+      QueryStatsParameters: *HmacParameters
+      RateLimitHz: *RateLimitHz
+      Suffix: "Hmac-Random-RateLimited"
+
+AutoRun:
+- When:
+    mongodb_setup:
+      $eq:
+      - standalone-query-stats-small-rate-limit
```

Comment:
I think given the name of the workload and the name of the actors, should we move this to match the other file? Just always run on the "normal" variants like "standalone" and "replica" and "shard"?

Reply:
That could also work. :shrug: This isn't specifically tied to any particular variant in practice, since we're manually resetting the rate limit at every turn. We could even put it into the same file and just give the rate-limited actors new names.